implement first pass at serialization

schuyler1d opened the pull request:
Well, where there's deserialization, some folks will need to serialize the other way :-/

Many thanks for your work on deserializing, and extensive test suite.

schuyler1d committed:
implement first pass at serialization

exhuma commented on storable/output.py:
```diff
@@ -0,0 +1,151 @@
+    """
+    negative = (value < 0)
+    value = abs(value)
+    assert(value < 128)
```
I would personally prefer a `ValueError` here. First, `assert` statements get dropped in "optimized mode" and have no helpful message associated with them. Both issues are resolved by throwing a `ValueError`.

exhuma reviewed (commented):
Just glanced over it. It looks good. I'll go over it in more detail in a second.

And thanks a lot for the PR 👍

exhuma commented:
I like these changes! We'll just have to figure out why the tests fail. If we get those fixed, I'll push a new release of the module.

schuyler1d committed:
inplace-modification method for serialized hashes

schuyler1d committed:
change asserts to value-errors

schuyler1d committed:
accomodate python3.5+ RecursionError

schuyler1d committed:
fix python3.3

schuyler1d committed:
add type to notimplementederror, or debugging

schuyler1d committed:
unicode literals for python2.7 support

schuyler1d committed:
basestring for python2.7 and pep8

schuyler1d committed:
support python2 pickier bytes() method by using bytearray

schuyler1d committed:
add testing for modify_hash and fix for py2.7

schuyler1d committed:
fix modify_hash for py2

schuyler1d committed:
python3/2 shenanigans

schuyler1d commented:
@exhuma ok, sorry for the commit-churn, (didn't have all the versions locally, so needed to do a little travis-poking).  now passing tests.

I also added a method I will need, `storable.output.modify_hash` -- it's experimental enough that I'm not sure it's worth documenting, but I added some tests.

exhuma commented:
Don't worry about the churn. As long as it's working and the tests pass, I'm fine with it ;)

I'll look it through tomorrow and will merge if possible.

schuyler1d commented:
a friendly ping

exhuma commented:
Has been merged and pushed to pypi as v1.1.0

Weirdly github did not recognize this as merge, so I'll have to close it manually. If you want you can resync your fork with this one so the "network" graph shows that everything is up-to-date.

I'm a bit afraid that a rebase which I had to do might have borked things up. And yet I was sure it wouldn't 😞 Hope it won't mess things up on your end.

schuyler1d commented:
Not at, all -- thanks for merging!
